src/reactor.h

```diff
@@ -0,0 +1,242 @@
+#ifndef CYCLUS_TRICYCLE_REACTOR_H_
+#define CYCLUS_TRICYCLE_REACTOR_H_
+
+#include <string>
+
+#include "cyclus.h"
+#include "boost/shared_ptr.hpp"
+
+namespace tricycle {
+
+/// @class Reactor
+///
+/// This Facility is intended
+/// as a skeleton to guide the implementation of new Facility
+/// agents.
+/// The Reactor class inherits from the Facility class and is
+/// dynamically loaded by the Agent class when requested.
+///
+/// @section intro Introduction
+/// Place an introduction to the agent here.
+///
+/// @section agentparams Agent Parameters
+/// Place a description of the required input parameters which define the
+/// agent implementation.
+///
+/// @section optionalparams Optional Parameters
+/// Place a description of the optional input parameters to define the
+/// agent implementation.
+///
+/// @section detailed Detailed Behavior
+/// Place a description of the detailed behavior of the agent. Consider
+/// describing the behavior at the tick and tock as well as the behavior
+/// upon sending and receiving materials and messages.
+class Reactor : public cyclus::Facility  {
+ public:
+  /// Constructor for Reactor Class
+  /// @param ctx the cyclus context for access to simulation-wide parameters
+  explicit Reactor(cyclus::Context* ctx);
+
+  /// The Prime Directive
+  /// Generates code that handles all input file reading and restart operations
+  /// (e.g., reading from the database, instantiating a new object, etc.).
+  /// @warning The Prime Directive must have a space before it! (A fix will be
+  /// in 2.0 ^TM)
+
+  #pragma cyclus
+
+  #pragma cyclus note {"doc": "A stub facility is provided as a skeleton " \
+                              "for the design of new facility agents."}
+
+  /// A verbose printer for the Reactor
+  virtual std::string str();
+
+  /// Set up policies and buffers:
+  virtual void EnterNotify();
+
+  /// The handleTick function specific to the Reactor.
+  /// @param time the time of the tick
+  virtual void Tick();
+
+  /// The handleTick function specific to the Reactor.
+  /// @param time the time of the tock
+  virtual void Tock();
+
+  #pragma cyclus var { \
+    "doc": "Nameplate fusion power of the reactor", \
+    "tooltip": "Nameplate fusion power", \
+    "units": "MW", \
+    "uilabel": "Fusion Power" \
```

Comment:
See comment below. Here are some example uitype implementations. Feel free to commit them if they're useful  or modify them if my suggestions aren't quite right. For example, assuming the power should be positive
```suggestion
    "uitype": "range", \
    "range": [0, 1e299], \
    "uilabel": "Fusion Power" \
```

Reply:
This is a good suggestion. I will go through and add these where they make sense. I just didn't think to do so initially.